### ISSUE TITLE
[Bug] `columnName` can be `undefined` in the `filter` object under `TableFilterRow.CellProps`

### ISSUE THREAD

**Issue by franklixuefei** [DevExpress/devextreme-reactive#1010]
- [x] I have searched this repository's [issues](https://github.com/devexpress/devextreme-reactive/issues) and believe that this is not a duplicate.

### Current Behaviour
when I was logging out the `filter` prop defined under `TableFilterRow.CellProps`, the `columnName` field is `undefined`, but if I was to pass a `filter` to `onFilter` callback defined under `TableFilterRow.CellProps`, I must of course pass the `columnName` value.

Please see the below snapshot for the value I got out of `filter` prop. **Note that only `value` field is defined.**

![image](https://user-images.githubusercontent.com/1334982/39388463-026d4afa-4a35-11e8-9bbc-509b74457b6c.png)

### Expected Behaviour

I think it is best to also pass the `columnName` value over to the user in the `filter` object, even though the column name can be derived from the `column` object. This will make the behavior more predictable and consistent.

This fix will also make the `Filter` interface in the type definition file more consistent, as below. Note that `columnName` prop is **required**.

```ts
// dx-react-grid.d.ts:266
/** Describes a filter. */
export interface Filter {
  /** Specifies the name of a column whose value is used for filtering. */
  columnName: string;
  /** Specifies the filter value. */
  value?: string;
}
```

### Your Environment

devextreme-reactive: 1.2.0-beta2
react: 16.2.0
browser: Chrome/Edge
material-ui: yes

**Comment by franklixuefei** [comment]
@kvet @gsobolev any thoughts on this? :-D

**Comment by gsobolev** [comment]
@franklixuefei,
 
We have just merged #1011. Thank you for your cooperation.
